### PR TITLE
363 Page Level Validation

### PR DESCRIPTION
**Pull request by mlauren**
## Description
- Adds onBlur functionality to some custom fields
- Removes unneeded second validation from some validator functions
- Moves form to higher level router level
- resets form `errors` and `touched` at page level `useEffect`

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#363

## Testing done
✅

## Screenshots
Screen Shot 2022-07-01 at 10 57 56 AM
Screen Shot 2022-07-01 at 10 58 15 AM
Screen Shot 2022-07-01 at 10 58 23 AM

## Definition of done
- [x] Documentation has been updated, if applicable
- [x] A link to the original github issue has been provided
- [x] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by bjmccotter7192** [comment]
After testing I found only one thing that was weird but I like this approach by disabling the button:

- PreviousNames page doesn't allow me to select the `No` option, when selected is still prompts you for the FullNameField 
- Style of the next button will need to be updated

**Comment by mlauren** [comment]
@bjmccotter7192 thanks for the feedback, I went ahead and fixed the style. @harshil1793 will be updating the form to no longer convert these values into booleans so I may wait until that part is merged in before merging this one.

**Review by micahchiang — COMMENTED** [review]
I think we might need to consider using conditional page navigation again, instead of relying on the disabled button attribute. VoiceOver doesn't alert the user of required fields when tabbing through and not filling anything out.

File: src/routing/Page.tsx
Comment (mlauren) [reply src/routing/Page.tsx:76]: Thanks for the feedback. I was adding this to test the functionality but didn't think to switch it back. I will look at erroring the page on button click.

Comment (micahchiang) [reply src/routing/Page.tsx:76]: I think there might be something weird going on with how our `TextField` component is rendering required errors. They are missing `role="alert"` which might be causing VoiceOver to skip over reading the error messages.

Comment (mlauren) [reply src/routing/Page.tsx]: Okay, I just altered it so it throws up errors if you try to submit. Would this be the intended functionality?

Comment (micahchiang) [reply src/routing/Page.tsx]: Yes. Change looks great, thanks @mlauren. PR looks fantastic as well. I'll create a bug ticket for the VoiceOver text field issue, since it's unrelated to your work here.

**Review by micahchiang — APPROVED** [review]

File: src/routing/Page.tsx
Comment (mlauren) [reply src/routing/Page.tsx]: @micahchiang is this something that's missing on the design system side or should be added to the input field? When I tab past the field in voiceover, it is reading "please provide a response" but it is the last thing that is read.

Comment (micahchiang) [reply src/routing/Page.tsx]: @mlauren - which field are you getting that on? In Firefox it isn't reading the error message for first or last name, but reads it for the radio group component on `ClaimantInformation`

**Pull request merged**
